spirv_msl.cpp

```diff
@@ -398,6 +510,19 @@ void CompilerMSL::emit_header()
 	statement("");
 	statement("using namespace metal;");
 	statement("");
+    emit_msl_defines();
+}
+
+void CompilerMSL::emit_msl_defines()
+{
+    statement("// Standard GLSL->MSL redefinitions");
+    statement("#define discard discard_fragment()");
```

Comment:
I think these things can be done by making CompilerGLSL::emit_instruction virtual in CompilerGLSL. Then MSL can implement the opcodes it needs to handle special, and fallback to parent class implementation for everything else.